api/src/main/java/io/kafbat/ui/config/auth/SimpleOAuthProxyConfig.java

```diff
@@ -0,0 +1,53 @@
+package io.kafbat.ui.config.auth;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.reactive.ReactorClientHttpConnector;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.netty.http.client.HttpClient;
+
+@Slf4j
+@Configuration
+@ConditionalOnProperty(value = "auth.type", havingValue = "OAUTH2")
+@EnableConfigurationProperties(SimpleOAuthProxyConfig.ProxyProperties.class)
+public class SimpleOAuthProxyConfig {
+
+  @Data
+  @ConfigurationProperties("auth.oauth2.proxy")
+  public static class ProxyProperties {
+    private boolean enabled = false;
+    private String host;
+    private Integer port;
+  }
+
+  @Bean(name = "oauth2WebClient")
+  @ConditionalOnProperty(value = "auth.oauth2.proxy.enabled", havingValue = "true")
```

Comment:
why do we need to support a proxy via custom settings rather than using a default system one provided by jre? Another point, in case we'd need a proxy somewhere else, we'd need to create yet another configuration with another bunch of properties to support proxy in multiple places.